### PR TITLE
eye kernel with tests

### PR DESCRIPTION
**Pull request by mchao409**
Added eye kernel (identity)

### CODE REVIEW

File: hammerblade/torch/kernel/kernel_eye.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#include <kernel_common.hpp>
+
+
+extern "C" {
+
+  __attribute__ ((noinline)) int tensorlib_eye(
+    hb_tensor_t* output, long* n, long* m) {
+    auto y = HBTensor<float>(output);
+    auto N = *n;
```
Comment (cbatten) [inline hammerblade/torch/kernel/kernel_eye.cpp]: I general we should not abuse auto. The type `long` is the same number of characters as `auto`. So this should probably be:
```
HBTensor<float> y(output);
long N = *n;
```
The above is more explicit, more compact, and more readable. Just a thought :)
```

File: aten/src/ATen/native/hammerblade/Eye.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+namespace native {
+
+  Tensor& eye_out_hb(Tensor& output, long n, long m) {
+    std::vector<eva_t> device_args;
+    std::vector<eva_t> device_ptrs;
+    if(m == -1) {
+      m = n;
+    }
+    output.resize_({n,m});
+    device_args.push_back(create_device_tensor(output, device_ptrs));
```
Comment (yodada) [inline aten/src/ATen/native/hammerblade/Eye.cpp]: I think you don't have to call these low level APIs yourself? `hb_offload_kernel` will take care of them for you.
You can write
```C
hb_offload_kernel(output, n, m, "tensorlib_eye");
```

File: hammerblade/torch/kernel/kernel_eye.cpp (hunk 1 above)
Comment (mchao409) [reply hammerblade/torch/kernel/kernel_eye.cpp]: Just made the change

File: aten/src/ATen/native/hammerblade/Eye.cpp (hunk 2 above)
Comment (mchao409) [reply aten/src/ATen/native/hammerblade/Eye.cpp]: Fixed!

**Review by yodada — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**